examples/sqlite/src/web/protected.rs

```diff
@@ -16,14 +16,7 @@ pub fn router() -> Router<()> {
 mod get {
     use super::*;
 
-    pub async fn protected(auth_session: AuthSession) -> impl IntoResponse {
-        match auth_session.user {
-            Some(user) => ProtectedTemplate {
-                username: &user.username,
-            }
-            .into_response(),
-
-            None => StatusCode::INTERNAL_SERVER_ERROR.into_response(),
-        }
+    pub async fn protected(Extension(User { username, .. }): Extension<User>) -> impl IntoResponse {
+        ProtectedTemplate { username }
```

Comment:
Thanks for considering this request!
The boilerplate reduction like in the example is what I'm hoping to reduce, especially when one has a large number of protected routes.

Reply:
Similar or better reduction should also be achievable by implementing FromRequest for your concrete user type: this would find AuthSession on the request extensions and then return the user directly (which also means you don't need to use Extension).

This might be a better solution, potentially.

Reply:
For example, that could look like this: https://github.com/maxcountryman/axum-login/commit/1717ead4ba1228807d27191e342e39f549e2ae9c

Reply:
This is indeed a better solution. Thanks @maxcountryman!